irmaclient/session.go

```diff
@@ -745,7 +745,12 @@ func (session *session) finish(delete bool) bool {
 func (session *session) fail(err *irma.SessionError) {
 	if session.finish(true) && err.ErrorType != irma.ErrorKeyshareUnenrolled {
 		irma.Logger.Warn("client session error: ", err.Error())
-		err.Err = errors.Wrap(err.Err, 0)
+		// Don't use errors.Wrap() if err.Err == nil, otherwise we may get
+		// https://yourbasic.org/golang/gotcha-why-nil-error-not-equal-nil/.
+		// since errors.Wrap() returns an *errors.Error.
+		if err.Err != nil {
+			err.Err = errors.Wrap(err.Err, 0)
+		}
```

Comment:
Did you manage to reproduce an error related to this fix, such that I can validate that this actually works?

Reply:
A little bare-bones, but you can artificially make your local IRMA server always reject everything as follows:

```diff
diff --git a/server/irmaserver/handle.go b/server/irmaserver/handle.go
index 296263ab..ef30f5ed 100644
--- a/server/irmaserver/handle.go
+++ b/server/irmaserver/handle.go
@@ -146,6 +146,8 @@ func (session *session) handlePostDisclosure(disclosure *irma.Disclosure) (*irma
 		rerr = session.fail(server.ErrorUnknown, err.Error())
 	}

+	rerr = session.fail(server.ErrorInvalidProofs, "go away")
+
 	return &irma.ServerSessionResponse{
 		SessionType:     irma.ActionDisclosing,
 		ProtocolVersion: session.Version,
```